docker-compose.yml

```diff
@@ -0,0 +1,61 @@
+version: '3.7'
+
+services:
+  mysql:
+    image: mysql:8.0
+    command: --default-authentication-plugin=mysql_native_password
+    environment:
+      MYSQL_ROOT_PASSWORD: password
+    volumes:
+      - mysql:/var/lib/mysql
+
+  postgres:
+    image: postgres:13.2
+    environment:
+      POSTGRES_USER: root
+      POSTGRES_PASSWORD: password
+    volumes:
+      - postgres:/var/lib/postgresql/data
+
+  app:
+    build:
+      context: .dockerdev
+      dockerfile: Dockerfile
+      args:
+        RUBY_VERSION: "2.7.2"
+        PG_VERSION: 13
+        NODE_VERSION: 14
+        MYSQL_VERSION: "8.0"
+        BUNDLER_VERSION: 2
+    image: solidus-3.0.0.rc1
+    command: bash -c "(bundle check || bundle) && tail -f Gemfile"
```

Comment:
`tail -f Gemfile`? 🤔 
I wonder the meaning of this line.

Reply:
It's in order to keep the container alive. If I omit `tail...` the process will exit when bundle ends its job. This way I can run `docker-compose exec app ...` and keep the history. Nothing prevents me to do `docker-compose run --rm app..` if I prefer it.

Reply:
Checking your README.md changes it looks like you do it to be able to use `docker-compose exec app ...` without a running app, right?

From my standpoint, this line looks like a workaround to keep the process up 🤔 but I haven't a better solution to offer here to keep the same behavior ATM.
What I see often is just to have `rails s` in `command`, which means that you need always a running app to be able to exec something else...

Reply:
Yeah, you need a live process to be able to `exec` on the container. As we can't just assume that the sandbox application exists, we need that trick.

Reply:
Having `tty: true`, another option could be to remove the `command` param.
(but perhaps an ENTRYPOINT is needed in your Dockerfile, not sure)

Reply:
I tried and then the running process becomes `irb`, which I guess is defined as entrypoint by the base ruby image. But doing so we would lose the option to automatically `bundle` when waking up, which is something I like. I can change `tail -f Gemfile` to `irb` if you prefer. Sure it's a more expensive process than `tail`, but I have no strong opinion here.

Reply:
If we are going to replace the command line with another blocking process I suppose that it could be fine with your solution anyway.
What about adding a RUN in the Dockfile to exec `bundle install` instead?

Reply:
I used to do that but it's a hassle as you need to rebuild the image every time you add a new gem to the Gemfile. You can still do `bundle` in a container that has some process running on it, but then when you remove the container for some reason and wake up the project again the application won't work until you realize that you have to rebuild again! It breaks with the flow you would use outside of docker.

Another option I've used sometimes is to just `bundle` neither on Dockerfile nor docker-compose, and then require it to be done manually, which is just the expected flow outside of docker. But having it done automatically and being able to have a development environment just after a single `docker-compose up -d` is something that is awesome (and anyway it doesn't prevent you to manually run `bundle` so it doesn't break any expectation).

Reply:
Ah ok - BTW it's up to you @waiting-for-dev, if you prefer the tail way then it's fine. At the moment I haven't a better solution if we want also the bundle already executed

Reply:
Cool. I made a change, though. I added a commit changing `tail -f Gemfile` to `tail -f /dev/null`. It's better because it doesn't pollute the logs and it communicates slightly better.